hydrolib/core/dflowfm/extold/models.py

```diff
@@ -0,0 +1,417 @@
+from enum import Enum, IntEnum
+from pathlib import Path
+from typing import Any, Callable, Dict, List, Optional, Union
+
+from pydantic import Field, root_validator, validator
+
+from hydrolib.core.basemodel import (
+    BaseModel,
+    DiskOnlyFileModel,
+    ModelSaveSettings,
+    ParsableFileModel,
+    SerializerConfig,
+)
+from hydrolib.core.dflowfm.extold.parser import Parser
+from hydrolib.core.dflowfm.extold.serializer import Serializer
+from hydrolib.core.dflowfm.polyfile.models import PolyFile
+
+
+class Quantity(str, Enum):
+    """Enum class containing the valid values for the boundary conditions category
+    of the external forcings.
+    """
+
+    # Boundary conditions
+    WaterLevelBnd = "waterlevelbnd"
+    """Water level"""
+    NeumannBnd = "neumannbnd"
+    """Water level gradient"""
+    RiemannBnd = "riemannbnd"
+    """Riemann invariant"""
+    OutflowBnd = "outflowbnd"
+    """Outflow"""
+    VelocityBnd = "velocitybnd"
+    """Velocity"""
+    DischargeBnd = "dischargebnd"
+    """Discharge"""
+    RiemannVelocityBnd = "riemann_velocitybnd"
+    """Riemann invariant velocity"""
+    SalinityBnd = "salinitybnd"
+    """Salinity"""
+    TemperatureBnd = "temperaturebnd"
+    """Temperature"""
+    SedimentBnd = "sedimentbnd"
+    """Suspended sediment"""
+    UXUYAdvectionVelocityBnd = "uxuyadvectionvelocitybnd"
+    """ux-uy advection velocity"""
+    NormalVelocityBnd = "normalvelocitybnd"
+    """Normal velocity"""
+    TangentialVelocityBnd = "tangentialvelocitybnd"
+    """Tangentional velocity"""
+    QhBnd = "qhbnd"
+    """Discharge-water level dependency"""
+    TracerBnd = "tracerbnd"
+    """User-defined tracer"""
+
+    # Meteorological fields
+    WindX = "windx"
+    """Wind x component"""
+    WindY = "windy"
+    """Wind y component"""
+    WindXY = "windxy"
+    """Wind vector"""
+    AirPressureWindXWindY = "airpressure_windx_windy"
+    """Atmospheric pressure and wind components"""
+    AirPressureWindXWindYCharnock = "airpressure_windx_windy_charnock"
```

Comment:
Just in case someone else also wants to review this PR. The `airpressure_windx_windy_charnock` is (among others) not listed in the manual, but in the issue description.